client/src/lib/data.ts

```diff
@@ -36,19 +38,24 @@ export async function fetchTickets(userId: number): Promise<Event> {
 }
 
 export async function purchaseTickets(
-  userId: number,
   eventId: number,
+  userId: number,
   ticketCount: number
 ): Promise<Event> {
+  const auth = localStorage.getItem(tokenKey);
+  const a = JSON.parse(auth as string);
   const req = {
-    method: 'GET',
+    method: 'POST',
     headers: {
       'Content-Type': 'application/json',
-      Authorization: `Bearer ${sessionStorage.getItem('token')}`,
+      Authorization: `Bearer ${a.token}`,
     },
-    body: JSON.stringify({ userId, eventId, ticketCount }),
+    body: JSON.stringify({ eventId, userId, ticketCount }),
   };
-  const res = await fetch(`/api/userEvents/${userId}`, req);
+  const res = await fetch(
+    `/api/userEvents/${eventId}/${userId}/${ticketCount}`,
```

Comment:
I can't see your server code in this PR, but it's a bit strange that you put eventId, userId, and ticketCount in the URL but also put them in the body. Usually you'd have them in only one place.

Which brings me to endpoint design. The idea behind the endpoint is that the URL identifies a resource (usually a row in a table). You would usually put the primary key in the URL. Then all the data that goes into that row is put in the request body.

This one is a little strange since you are updating a join table, but it makes sense to have the URL be '/api/userEvents/:eventId/:userId' since that identifies a single row. Then you would put ticketCount in the body.

By the way, I can't see your schema.sql file, but I assume ticketCount is _not_ part of the primary key for the userEvents table. It shouldn't be, but I'm just making sure.